turingschool-examples/whats-cookin-starter-kit
# Feature/user class test suite

## Is this a fix or a feature?

Testing and refactoring, no added functionality

## What is the change?

- Adds a test suite for the User class testing functionality that's been added thus far for iteration 2. Additional tests will be added with new functionality required for iteration 3, but this at least gets us started.
- Makes changes to the User class to facilitate testing - all localStorage functionality was moved out of the class and into scripts.js since the testing environment doesn't recognize localStorage
- Removes the search methods on the User class and reworks search functionality to rely solely on the RecipeRepository. Instead of the User class using its own methods to filter its favorites array, the program conducts a master search of the RecipeRepository and filters out results that don't appear in the currentUser's favorites library.

## Where should the reviewer start?

See User-test.js in the test file

User.js
- Search methods removed
- localStorage functionality in properties for favoriteRecipes and RecipesToCook has been moved to scripts.js
- localStorage functionality in remaining methods was also delegated to scripts.js

Scripts.js
- line 28 new function loadRandomUser, invoked on page load (see eventListener line 219). For development purposes a random user is not currently being randomly selected on page load. This line of code will be replaced by the comments upon project submission.
- line 36, helper function helps load localStorage data passing it directly into the new User instantiation
- line 83 - updates localStorage whenever recipes are favorited/unfavorited
- lines 162-186 - updated favorites search - executes master search method on recipe repository and then filters our results that appear in the users favorites library.

## How should this be tested?

Load application in browser and test all features - UI should function exactly the same as the last PR.

Run npm test on the test file to verify all tests being passed (you'll need to un-comment the require statements in the class files). Review test suites.
